File: 800/Do Not Be Distracted!.cpp
#include <iostream>
#include <string>
#include <set>

using namespace std;

int main()
{    
    // Number of test Cases
    int test_cases;
    cin >> test_cases;

    // Looping for Each Test Case
    for (int test_case = 0; test_case < test_cases; test_case++) {
        
        int len, idx;
        cin >> len;

        set <char> seen;
        char prev = ' ';


        string sus = "YES";

        for (idx = 0; idx < len; idx++) {
            char task;
            cin >> task;

            if (seen.count(task) != 0 and task != prev) {
                sus = "NO";
                break;
            } else {
                seen.insert(task);
            }
            prev = task;
        }

        for (int temp = len - 1 - idx; temp > 0; temp -- ) cin >> prev;

        cout << sus << endl;
    }
 
    return 0;
}

File: 800/Do Not Be Distracted!.py
import math

for case_idx in range(int(input())) :  

    len = int(input())
    case = input()

    check = set()
    prev = " "

    flag = "YES"

    for test in case: 
        if test in check and test != prev : 
            flag = "NO"
            break
        else:
            check.add(test)

        prev = test
        
    print(flag)